CMakeLists.txt

```diff
@@ -0,0 +1,179 @@
+cmake_minimum_required(VERSION 3.25...3.29)
```

Comment:
I have upgraded the minimum CMake to be in sync with CentOS stream 9 (should be ported to RHEL 9, I'll try to make some noise on that) and Ubuntu 24.04. Users could use `pip install cmake` and HPCs should get ready.